### ISSUE TITLE
tree-sitter movements unreliable in vue files

### ISSUE THREAD

**Issue by alexej-rotar** [helix-editor/helix#11299]
### Summary

When editting vue files, it used to be impossible to use tree-sitter movements within the `<script>` tags. I was very happy to learn that release 24.03 made this possible. Unfortunately, I found that it does not reliably work for me. In many cases it works just as expected, but in many cases it doesn't - i.e. it behaves the way it did before, meaning `A-o` expands to include the entire `<script>` tag rather than the parent in the JS/TS syntax tree. Unfortunately, it's very hard to predict when that happens. I even found cases where it consistently fails on some lines and consistently works on others. E.g. I have something like this:

```typescript
foo.map(x => {
  const bar = 'baz';
  return `foo is ${bar}`;
})
```

where tree-sitter motions *do not* work in the lines up to `const bar ...` but *do* work starting at `return ...`. I am very puzzled when that happens so it's hard to give you exact reproduction steps (unless I would paste the entire file in here, which I can't).

### Reproduction Steps

 

I tried this:

1. `hx example.vue`
2. navigate to the `<script>` region
3. press `A-o`

I expected this to happen:
*Always* expand selection to the closest parent in the JS/TS tree.

Instead, this happened:
*Sometimes* the selection is being expanded to the closest parent in the HTML tree, i.e. the entire `<script>` tag.

### Platform

Linux

### Terminal Emulator

alacritty

### Installation Method

pacman

### Helix Version

24.7

**Comment by pascalkuthe** [comment]
I can't reproduce this. I used the following these file (please post the whole thing, I needed to guess what you mean about the script thing):

```
<script>
foo.map(x => {
  const bar = 'baz';
  return `foo is ${bar}`;
})
</script>
```

`a-o` is working as expect here on the JS elements

**Comment by AS400JPLPC** [comment]
![Capture d’écran du 2024-07-26 00-12-21](https://github.com/user-attachments/assets/33686333-81da-48dd-b375-660858a9d298)
![Capture d’écran du 2024-07-26 00-55-35](https://github.com/user-attachments/assets/bfa9e62e-d46d-40c5-8419-5cb50f7ad261)

**Comment by alexej-rotar** [comment]
Thanks for the quick reply! I can't (am not allowed to) post the exact file here. The thing I posted was just roughly resembling it, and might have worked for me as well. As I said, I didn't really know, when things work and when they don't. However, I was able to strip down one of the files that don't work down to two examples that consistently fail for me:

```vue
<script setup lang="ts">
const foo = 5;
// some comment
const bar = foo;
</script>
```

Anything above the comment doesn't work, anything below it does (including the comment). Interestingly, if I run `:tree-sitter-subtree` on the line above the comment, it does output a tree rather than just a single text node. Second example:

```vue
<script setup lang="ts">
const foo = 'bar'.match(/foo/);
const bar = foo;
</script>
```

Anything before the regex `/foo/` does not work, anything after it (including the regex itself) works. Hope that helps!
